### ISSUE TITLE
JuiceFS gateway support for object metadata

### ISSUE THREAD

**Issue by stevapple** [juicedata/juicefs#5367]
**What would you like to be added**:
Add support for storing, fetching and modifying object metadata in JuiceFS gateway.

**Why is this needed**:
Object metadata is used for statistics and will be user-facing (as HTTP headers), compared to user tags. This is extremely useful when we want to expose some custom metadata to downstream users (eg. web clients).

**Comment by davies** [comment]
Can you provide a more detailed example on how this feature will be used?

**Comment by stevapple** [comment]
@davies Object metadata is tightly associated with the object, and is user-facing. If we want to allow end-users to provide and get metadata through a public URL (pre-signed or anonymous), this is the only choice. We’re especially using this feature to add MIME and other useful information for media/file management backed by S3.

**Comment by prOOrc** [comment]
We have encountered an issue because of rclone tries to store mtime in the object metadata.

**Comment by zhijian-pro** [comment]
@prOOrc  Can you provide the specific details for reproducing the error, we will not store the metadata, but it should not cause an error.

**Comment by zhijian-pro** [comment]
@stevapple  We have no support plan for object metadata at this time. 
Supporting the object metadata  can lead to degraded performance in getObject operations.

**Comment by prOOrc** [comment]
@zhijian-pro Yes, it doesn't cause an error. But rclone tries to preserve the mtime in metadata. We have to ignore the modification time during file synchronization from/to sources that support the mtime.
I added metadata support in my [fork](https://github.com/juicedata/juicefs/pull/5409). It can be enabled optionally.

**Comment by davies** [comment]
@prOOrc Can you send a PR for it?
